### PR TITLE
ftp: adds a config option ftp-hash for autofp-scheduler

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5205

Describe changes:
- Adds an option `ftp-hash` for `autofp-scheduler` : like `hash` except for FTP-ish flows

No S-V test as this is about a concurrency issue...

### CODE REVIEW

File: doc/userguide/performance/runmodes.rst [hunk 1]
```diff
@@ -46,3 +46,21 @@ useful during development.
...
+Suricata may use different ways to load balance the packets to process
+between different threads with the configuration option `autofp-scheduler`.
+
+The default value is `hash`, which means the packet is assigned to threads
+using the 5-7 tuple hash, which is also used anyways to store the flows
+in memory.
+
+This option can also be set to
+- `ippair` : packets are assigned to threads using addresses only.
+- `ftp-hash` : same as `hash` except for flows that may be ftp or ftp-data
```
Comment (jasonish) [inline doc/userguide/performance/runmodes.rst:62]: Could this be true for any protocol that has a control flow and a data flow? SIP comes to mind, though I don't think we do anything with the data path there (yet). I wonder if a better name is `related` (taken from iptables terminology)?

Any reason this is not done by default?

Comment (catenacyber) [reply doc/userguide/performance/runmodes.rst:62]: Who can test ?

And is this the right place to implement such a change ?

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 6663

**Comment by suricata-qa** [comment]
Warning: no commits in this PR have specified the following ticket(s):
- 5205 - https://redmine.openinfosecfoundation.org/issues/5205

Please update the commit(s) and submit a new PR.

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7319

**Pull request closed without merging**